### PR TITLE
gh-3063 Add missing relay tracking

### PR DESCRIPTION
**Pull request by elgatovital**
Handles #3063

Changes proposed in this pull request:
- Add tracking events for success screen
- Events indicating submission status to relay service
- Distinguish btw screen and user action events

### CODE REVIEW

File: Multisig/UI/Transaction/ExecuteTransaction/SuccessViewController.swift [hunk 1]
```diff
@@ -21,26 +21,29 @@ class SuccessViewController: UIViewController {
...
+        secondaryAction: String?
     ) {
         self.init(nibName: nil, bundle: nil)
         self.titleText = titleText
         self.bodyText = bodyText
         self.primaryAction = primaryAction
         self.secondaryAction = secondaryAction
+    }
+
+    func setTrackingData(trackingEvent: TrackingEvent?, trackingParams: [String: Any]? = nil) {
```
Comment (MouazAlzahabi) [inline Multisig/UI/Transaction/ExecuteTransaction/SuccessViewController.swift:46]: Why this has been moved to new method, we can still keep it in the initializer and I would keep it there

Comment (elgatovital) [reply Multisig/UI/Transaction/ExecuteTransaction/SuccessViewController.swift:46]: controller can be created without tracking info as it is optional and not essential. Also this way decision about tracking parameters can be delayed.

Comment (elgatovital) [reply Multisig/UI/Transaction/ExecuteTransaction/SuccessViewController.swift:46]: additionally with this method event and it's parameters are bundled together now.

File: Multisig/UI/Transaction/ExecuteTransaction/ReviewExecutionViewController.swift [hunk 2]
```diff
@@ -633,9 +633,11 @@ class ReviewExecutionViewController: ContainerViewController, PasscodeProtecting
                 switch result {
                 case .failure(let error):
                     self.submitButton.state = .normal
+                    Tracker.trackEvent(.relayUserFailure)
                     self.didSubmitFailed(error)
 
                 case .success:
+                    Tracker.trackEvent(.relayUserSuccess)
```
Comment (biafra23) [inline Multisig/UI/Transaction/ExecuteTransaction/ReviewExecutionViewController.swift:640]: What is the difference between `.relayUserSuccess` and `.successTxRelay`? Do we need to send both?

Comment (biafra23) [reply Multisig/UI/Transaction/ExecuteTransaction/ReviewExecutionViewController.swift:640]: Apparently `.successTxRelay` is a screen event and `.relayUserSuccess` is a user event. 

Do we need the user event?

Comment (elgatovital) [reply Multisig/UI/Transaction/ExecuteTransaction/ReviewExecutionViewController.swift:640]: we don't show success screen after safe creation. Screen event `.successTxRelay` is used when success screen is shown, `.relayUserSuccess` is for successful submission to the relay service. For safe creation we have only user events (also when signed key is used). But for other transactions, asset transfer, and claiming we show the success screen with corresponding tracking events. To be consistent I have added screen event also for relaying.

**Review by biafra23 — APPROVED** [review]
LGTM 👍🏻

File: Multisig/UI/Transaction/ExecuteTransaction/SuccessViewController.swift (hunk 1 above)
Comment (MouazAlzahabi) [reply Multisig/UI/Transaction/ExecuteTransaction/SuccessViewController.swift:46]: I would actually still prefer to find a way to do all in initializer, better than two calls every time, as you can see in the places we currently use, but if you want to keep it like this then fine

**Pull request merged**